crates/next-core/src/next_font_google/mod.rs

```diff
@@ -156,17 +157,28 @@ impl ImportMappingReplacement for NextFontGoogleCssModuleReplacer {
         )
         .await?;
 
-        // TODO(WEB-274): Emit an issue instead
-        if stylesheet_res.status >= 400 {
-            bail!("Expected a successful response for Google fonts stylesheet");
-        }
+        let stylesheet = match &*stylesheet_res {
+            FetchResult::Ok(r) => Some(r.await?.body.to_string().await?.clone()),
+            FetchResult::Err(err) => {
+                // Inform the user of the failure to retreive the stylesheet, but don't
+                // propagate this error. We don't want e.g. offline connections to prevent page
+                // renders during development. During production builds, however, this error
+                // should propagate.
+                //
+                // TODO(WEB-283): Use fallback in dev in this case
+                // TODO(WEB-293): Fail production builds (not dev) in this case
+                err.to_issue(IssueSeverity::Warning.into(), css_virtual_path)
+                    .as_issue()
+                    .emit();
+
```

Comment:
Since this permanently caches the results (the error is handled), this means that if the user transitions from offline to online, they'll always get this empty/fallback result. I'd like to be able to invalidate the task here  (`turbo_tasks::get_invalidator().invalidate();`) so it's retried later, but it causes the request for the document to be held open indefinitely. Any ideas?

Reply:
hmmm, maybe it would be possible to add an invalidator to the `FetchResult`, but that also seems a bit hacky

Reply:
cc @sokra